store/src/state.rs

```diff
@@ -415,21 +447,16 @@ fn block_to_nullifier_data(block: BlockNumber) -> Word {
     [Felt::new(block as u64), Felt::ZERO, Felt::ZERO, Felt::ZERO]
 }
 
-// /// Returns the block number encoded as a leaf value to be used in the TSMT.
-// fn nullifier_data_to_block(block: Word) -> BlockNumber {
-//     block[0].as_int() as BlockNumber
-// }
-
 /// Creates a [SimpleSmt] tree from the `notes`.
-pub fn build_notes_tree(notes: &[Note]) -> Result<SimpleSmt, anyhow::Error> {
+pub fn build_notes_tree(notes: &[NoteCreated]) -> Result<SimpleSmt, anyhow::Error> {
     // TODO: create SimpleSmt without this allocation
     let mut entries: Vec<(u64, Word)> = Vec::with_capacity(notes.len() * 2);
 
-    for (index, note) in notes.iter().enumerate() {
+    for note in notes.iter() {
         let note_hash = note.note_hash.clone().ok_or(StateError::MissingNoteHash)?;
         let account_id = note.sender.try_into().or(Err(StateError::InvalidAccountId))?;
         let note_metadata = NoteMetadata::new(account_id, note.tag.into(), note.num_assets.into());
-        let index = (index as u64) * 2;
+        let index = note.note_index as u64;
```

Comment:
This was likely a bug.

I have not added a duplicate check because `SimpleSmt` constructor has one already.